### DISCUSSION TITLE
data field in vega lite spec

### DISCUSSION THREAD

**Discussion by HaveF** [rclement/datasette-dashboards#168] [General]
Hi, @rclement  How have you been? 😄

I noticed a question that I had observed earlier, but it only occurred to me today, so I thought I'd ask. 

If a chart is created using vega-lite as the plotting method, it can be adjusted using the "Open in Vega Editor" option. 

![image](https://github.com/rclement/datasette-dashboards/assets/54462/4c943dba-2e6f-4dd5-9f61-5b1cf82cd69a)

For example, in [the demo](https://datasette-dashboards-demo.vercel.app/-/dashboards/job-offers-stats?date_start=2021-01-01), the data field for the graph is structured as follows:

```
  "data": {
    "values": [
      {"source": "Apec", "count": 980},
      {"source": "RegionsJob", "count": 394},
      {"source": "Indeed", "count": 20}
    ],
    "format": {"type": "json"}
  },
```

This approach is great as it allows for easy data observation and direct adjustment of the vega-lite specification. However, when dealing with a large amount of data, it becomes inconvenient as we have to scroll a long way past the data field in order to make adjustments to the specification. I recall that in a previous version, the data field was displayed in the form of a URL, which was much cleaner. Why has it changed to the current format?

If it were in URL format, it would be easier to use in other tools as well(of course we can get url from datasettle by ourselves). 

I'm curious about your thoughts on this matter. Thanks!

**Comment by rclement** [comment]
Hi @HaveF! You are very welcome about this question!
Indeed, this change occurred transparently with release [version 0.6.1](https://github.com/rclement/datasette-dashboards/blob/master/CHANGELOG.md#060---2023-08-18) to allow for chart row limit tooltip. Here is why:
- In order to obtain the `truncated` field to display a row limit tooltip on a chart, the JSON API request is mandatory
- For most chart types, it was a no-brainer: simply switch the API request format to get the extra attribute then pass along the data to the chart
- For Vega/Vega-Lite, I had to make a choice: either standardize data API calls like all charts and pass along the data (1 request needed), or keep the current model of passing the data URL at the expense of having one more API request to get the `truncated` attribute
- I chose to optimize the number of API calls for Vega/Vega-Lite and standardize the data retrieval model for all charts

In any case, charts are usually for aggregated data so you should not need to display more than a hundreds data points so I still is usable. Also, Vega Editor works best with inline data: I had trouble sometimes with data URL no showing properly the data in the debug tools.

I hope that clarifies the change.

**Reply by HaveF** [reply]
For facet views, it easy to use hundreds data points 😂 😂 😂 

But I see your points! Thanks for the clarification ❤️
